colossalai/engine/ophooks/_shard_grad_ophook.py

```diff
@@ -0,0 +1,31 @@
+import torch
+from colossalai.registry import OPHOOKS
+
+from . import BaseOpHook
+
+
+@OPHOOKS.register_module
+class ShardGradHook(BaseOpHook):
```

Comment:
@feifeibear Need a mechanism to guarantee different execution orders. Not related to this PR